src/Qowaiv/Financial/IbanParser.cs

```diff
@@ -12,24 +14,42 @@ internal static partial class IbanParser
         var index = 0;
         var pos = 0;
         var id = 0;
+        var prefixed = false;
 
-        foreach (var ch in str)
+        while (index < str.Length)
```

Comment:
The while loop is a bit complicated. Am wondering is the following idea is possible:
- Loop till no markup
- Check if there is one IBAN prefix (either (IBAN) or IBAN, first 2 chars would be enough to call function) -> move to after prefix
- Loop till no markup
- Use Bban to parse rest of string

Not sure if this would prevent a double prefix.

Reply:
Your approach would slow down all parsing, while prefixes are relatively uncommon. With this approach,  only if the default flow does not result in a valid IBAN (a valid country code to start with), a prefix check kicks in. It also does not require to check the string multiple times or creating substrings.

Reply:
I do think the above would only require one passthrough, but perhaps it would not optimize default case.
Also I think that `str[(index - 2)..]` and `str[(index - 1)..]` do create copies in the current implementation.